src/libsyntax/visit.rs

```diff
@@ -659,6 +659,7 @@ pub fn walk_expr<'a, V: Visitor<'a>>(visitor: &mut V, expression: &'a Expr) {
         ExprKind::Struct(ref path, ref fields, ref optional_base) => {
             visitor.visit_path(path, expression.id);
             for field in fields {
+                walk_list!(visitor, visit_attribute, field.attrs.iter());
```

Comment:
It cannot. `ThinVec` does not implement `IntoIterator`. `&ThinVec` also does not, leaving only this approach.